subsys/bluetooth/audio/pacs.c

```diff
@@ -256,6 +258,20 @@ static int set_available_contexts(uint16_t contexts, uint16_t *available,
 	return 0;
 }
 
+static int set_supported_contexts(uint16_t contexts, uint16_t *supported)
```

Comment:
What if the current available contexts contains bits that are no longer in our supported contexts? I can't remember if any of the upper layers (e.g. CAP) specifies what to do in that case.

Reply:
We might change available contexts here without notification.

Reply:
Not sure what you mean by that - Can you elaborate?

Reply:
Correct me if I'm thinking wrong but shouldn't we change available contexts here also? When we try to change supported contexts and they are different from currently available contexts we shall "disable" available contexts that are no longer valid.

Reply:
I agree. 

Since we are restricting the `available_contexts` to only the bits in `supported_contexts`, a change in the `supported_contexts` should also remove any bits that no longer overlaps between the two values (i.e. removes contexts from available that are not in supported). 

If the available contexts change as part of this, we should also notify that value. 

What should happen if we have a stream in the streaming state with context_A, and we remove context_A here? Should we reject the request from the upper layers? Should we stop the stream? Do nothing?

Reply:
Is the current code doing this properly in your opinion? Regarding the streaming - I think this is the topic for a different PR.

Reply:
See https://github.com/zephyrproject-rtos/zephyr/pull/52530#discussion_r1037967001